src/main/java/frc/robot/subsystems/ArmControlSubsystem.java

```diff
@@ -0,0 +1,312 @@
+import frc.robot.constants.ArmConstants;
+import frc.robot.constants.CompConstants;
+
+
+public class ArmControlSubsystem extends SubsystemBase {
+
+  public enum ArmMotorMode {
+    COAST,
+    BRAKE,
+    OFF
+  }
+
+  public enum MoveArmConfig {
+    SIMULTANEOUS,
+    SEQ_EXTEND_THEN_PIVOT,
+    SEQ_PIVOT_THEN_EXTEND
+  }
+  
+  
+  private final WPI_TalonFX mLeftPivotController = new WPI_TalonFX(ArmConstants.LEFT_PIV_MOTOR_ID);
+  private final WPI_TalonFX mRightPivotController = new WPI_TalonFX(ArmConstants.RIGHT_PIV_MOTOR_ID);
+  private final CANSparkMax mExtensionController = new CANSparkMax(ArmConstants.EXT_SPARK_ID, MotorType.kBrushless);
+
+  private final DutyCycleEncoder mAbsPivEncoder = new DutyCycleEncoder(ArmConstants.DIOPORTPIV);
+  private final RelativeEncoder mExtensionEncoder = mExtensionController.getEncoder();
+
+  private final SlewRateLimiter mPivotRateLimiter;
+  private final PIDController mExtensionPID;
+  private final PIDController mPivotPID;
+
+  // This is called Ultra Instinct because setting this boolean to true removes the clamps
+  // so that if the belt skips, the operator can still run manually without the bad offsets
+  private boolean mUltraInstinct = false;
+
+  private double mPivotRelEncoderOffsetRot;
+  private double mCurrentPivotRotation;
+  private double mDesiredPivotRotation = ArmConstants.MIN_PIV_ANGLE_RAD;
+
+  private double mCurrentExtensionDistance = ArmConstants.ZERO_EXTENSION_IN;
+  private double mDesiredExtensionDistance = ArmConstants.MIN_EXT_LEN_IN;
+
+
+
+  private boolean mIsInitialized = false;
+
+  private final SendableChooser<ArmMotorMode> mChooser = new SendableChooser<>();
+
+
+  public ArmControlSubsystem() {
+
+    mPivotPID = new PIDController(1.4, 0, 0);
+    mPivotPID.setTolerance(ArmConstants.RESTING_PIV_TOLERANCE_RAD);
+
+    mExtensionPID = new PIDController(0.19, 0, 0);
+    mExtensionPID.setTolerance(ArmConstants.RESTING_EXT_TOLERANCE_IN);
+
+    mPivotRateLimiter = new SlewRateLimiter(ArmConstants.MAX_PIV_RATE_RAD_SEC);
+
+    mAbsPivEncoder.setConnectedFrequencyThreshold(ArmConstants.CONNECTION_THRESH_HZ);
+   
+    this.mPivotRelEncoderOffsetRot = -mAbsPivEncoder.getAbsolutePosition() * ArmConstants.PIVOT_ABS_ENC_TO_ROTATION + ArmConstants.PIV_INIT_OFFSET_ROT;
+   
+    setDefaultConfig();
+
+    SmartDashboard.putNumber("kG", ArmConstants.KG);
+    
+    mChooser.addOption("Brake", ArmMotorMode.BRAKE);
+    mChooser.addOption("Coast", ArmMotorMode.COAST); 
+    mChooser.setDefaultOption("Brake", ArmMotorMode.BRAKE);
+
+    SmartDashboard.putData("ArmMotorMode", mChooser);
+    
+  }
+
+  public void setDefaultConfig(){
+    mRightPivotController.configFactoryDefault();
+    mLeftPivotController.configFactoryDefault();
+
+    mRightPivotController.follow(mLeftPivotController);
+
+    mRightPivotController.setInverted(TalonFXInvertType.OpposeMaster);
+    mLeftPivotController.setInverted(ArmConstants.LEFT_PIV_MOTOR_INVERTED);
+
+    this.updateModes();
```

Comment:
Do a search and replace of your repo to get rid of all the 'this.' in the code base.

Reply:
Done. The one exception is for the lamdas in the Arm subsystem which require the keyword.